XamarinCommunityToolkit/Behaviors/NumericValidationBehavior.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xamarin.Forms;
+
+namespace XamarinCommunityToolkit.Behaviors
+{
+    public class NumericValidationBehavior : BaseBehavior<Entry>
+    {
+        public static readonly BindableProperty TextColorProperty = BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(NumericValidationBehavior));
+
+        public Color TextColor
+        {
+            get => (Color)GetValue(TextColorProperty);
+            set => SetValue(TextColorProperty, value);
+        }
+
+        protected override void OnAttachedTo(Entry entry)
+        {
+            entry.TextChanged += OnEntryTextChanged;
+            base.OnAttachedTo(entry);
+        }
+
+        protected override void OnDetachingFrom(Entry entry)
+        {
+            entry.TextChanged -= OnEntryTextChanged;
+            base.OnDetachingFrom(entry);
+        }
+
+        void OnEntryTextChanged(object sender, TextChangedEventArgs args)
+        {
+            double result;
+            var isValid = double.TryParse(args.NewTextValue, out result);
+            ((Entry)sender).TextColor = isValid ? Color.Default : (TextColor != null ? TextColor : Color.Red);
```

Comment:
I wonder what happens if people might have a custom text color fitting their styling. Let's say you have a gray tone as text color. So if we add this behavior and always set it to `Color.Default` for them if valid it would override their own styling. Perhaps we should store the current text color in a private variable on attaching and set it to that when valid?

Reply:
I will review the logic tonight.